### ISSUE TITLE
Bug: [no-throw-literal] False positive report on `Readonly<Error>`

### ISSUE THREAD

**Issue by auvred** [typescript-eslint/typescript-eslint#8233]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.tsx&code=CYUwxgNghgTiAEYD2A7AzgF3gMyUgXPAEohTCoQCeAPAKIwxIwB8AUBgBaMDuOeQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1vwAtoOA7rUb5E0AIbx0UMf2iRwYAL4glQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
declare const foo: Readonly<Error>
throw foo
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-throw-literal": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Should not report on `Readonly<Error>`

### Actual Result

False positive report on `Readonly<Error>`

> Expected an error object to be thrown.

### Additional Info

Followup for https://github.com/typescript-eslint/typescript-eslint/pull/8011#discussion_r1442821388

**Comment by bradzacher** [comment]
In theory this is a bug but idk if it is actually something people will ever hit in practice.

Its just not a common thing to pass around immutable error instances - it's even rare to pass error instances around much!

**Comment by auvred** [comment]
Yes, I agree, but we already implemented this for [`prefer-promise-reject-errors`](https://main--typescript-eslint.netlify.app/rules/prefer-promise-reject-errors) rule ([playground](https://main--typescript-eslint.netlify.app/play/#ts=5.3.3&fileType=.ts&code=CYUwxgNghgTiAEYD2A7AzgF3gMyUgXPAEohTCoQCeAPAKIwxIwB8AUAAqMC2AlmiADo4AK3AYAFLiQBKVgHo58JcpUqAevCgQI8Pjh4oQrDtz6CRY8QHIpV6UA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQA7UQDNFoBafAewFsBLZRUggK0QGMAXUk6C6VDSBUKFIAGnDZIAAXYBPXClbQauTing0AduwD0%2BIiXI9a9RohYcu0Hn3RRuvSBIC%2BIZ0A&tsconfig=&tokens=false)). It would be nice to synchronize these rules.

Also, to support `Readonly<Error>` we just need to call `isReadonlyError` in this line
https://github.com/typescript-eslint/typescript-eslint/blob/c3767edf65716be08df25723d7dbb770de0e7037/packages/eslint-plugin/src/rules/no-throw-literal.ts#L83

So this is a pretty trivial fix!

**Comment by bradzacher** [comment]
7 months later and this request has received 0 community reactions.
As per our [workflow guidelines](https://typescript-eslint.io/maintenance/issues#community-engagement-evaluation) - we are taking this as signal that this check is not important to the broader community and are thus rejecting the proposal. Thanks for filing!
